docs/src/userguide/language_basics.rst

```diff
@@ -433,25 +433,63 @@ type declaration and let them be objects.
 Type qualifiers
 ---------------
 
-Cython supports ``const`` and ``volatile`` `C type qualifiers <https://en.wikipedia.org/wiki/Type_qualifier>`_::
+Cython supports ``const`` and ``volatile`` `C type qualifiers <https://en.wikipedia.org/wiki/Type_qualifier>`_
 
-    cdef volatile int i = 5
+.. tabs::
+
+    .. group-tab:: Pure Python
+
+        .. literalinclude:: ../../examples/userguide/language_basics/type_qualifiers.py
+
+
+
+    .. group-tab:: Cython
+
+        .. literalinclude:: ../../examples/userguide/language_basics/type_qualifiers.pyx
+
+
+Similar to pointers Cython supports shortcut types that can be used in pure python mode. The following table shows several examples:
+
+.. list-table:: Type qualifiers shortcut types
+   :widths: 100 10 10
+   :header-rows: 1
+
+   * - Cython
+     - Full Annotation type
+     - Shortcut type
+   * - ``const float``
+     - ``cython.const[cython.float]``
+     - ``cython.const_float``
+
+   * - ``const void *``
+     - ``cython.pointer[cython.const[cython.void]]``
+     - ``cython.p_const_void``
+
+   * - ``const int *``
+     - ``cython.pointer[cython.const[cython.int]]``
+     - ``cython.p_const_int``
+   * - ``int * const``
+     - ``cython.const[cython.pointer[cython.int]]``
+     - ``cython.const_p_int``
+   * - ``const int * const``
+     - ``cython.const[cython.pointer[cython.const[cython.int]]``
+     - ``cython.const_p_const_int``
+
+   * - ``uchar const *``
+     - ``cython.pointer[cython.const[cython.uchar]]``
+     - ``cython.const_p_uchar``
 
-    cdef const int sum(const int a, const int b):
-        return a + b
+   * - ``long const **``
```

Comment:
I, for one, had to look it up that C allows putting `const` before or after the type, and Cython requires it to go first. To avoid confusion for people who learn C coming from the Python side, I suggest using only the `const`-first notation in our documentation.
```suggestion
   * - ``const long **``
```